source/http_stream.c

```diff
@@ -23,6 +23,9 @@ struct http_stream_binding {
     napi_threadsafe_function on_body;
     struct aws_http_message *response; /* used to buffer response headers/status code */
     struct aws_http_message *request;
+
+    uint64_t pending_length; /* used to check all the body callbacks for nodejs has been invoked */
```

Comment:
"used to ensure that all of the body callbacks to node have been invoked"

Reply:
Also, looking at this, this value gets written to from both the node thread and the I/O threads. This should be an atomic.